File: HeapSort/HeapSort.py
import random

def randomnumber(a):
    for i in range(len(a)):
        a[i] = random.randint(1, 100)

def printarray(a):
    for i in range(len(a)):
        print(a[i], end=" ")
    print()

def heapify(a, n, i):
    largest = i
    left = 2 * i + 1
    right = 2 * i + 2

    if left < n and a[left] > a[largest]:
        largest = left

    if right < n and a[right] > a[largest]:
        largest = right

    if largest != i:
        a[i], a[largest] = a[largest], a[i]
        heapify(a, n, largest)

def HeapSort(a):
    n = len(a)

    for i in range(n // 2 - 1, -1, -1):
        heapify(a, n, i)

    for i in range(n - 1, 0, -1):
        a[i], a[0] = a[0], a[i]
        heapify(a, i, 0)
array = [0] * 15
randomnumber(array) 
longitud = len(array)
print("Arreglo Original: ")
printarray(array)
HeapSort(array)
print("Array Ordenado:")
printarray(array)

File: HeapSort/HeapSort.cs
using System;

class Program
{

    static void randomnumber(int[] a)
    {
        Random rand = new Random();
        for (int i = 0; i < a.Length; i++)
        {
            a[i] = rand.Next(1, 100);
        }
    }

    static void printarray(int[] a)
    {
        for (int i = 0; i < a.Length; i++)
        {
            Console.Write(a[i] + " ");
        }
        Console.WriteLine();
    }

    static void heapify(int[] arr, int n, int i)
    {
        int largest = i;
        int left = 2 * i + 1;
        int right = 2 * i + 2;

        if (left < n && arr[left] > arr[largest])
            largest = left;

        if (right < n && arr[right] > arr[largest])
            largest = right;

        if (largest != i)
        {
            int swap = arr[i];
            arr[i] = arr[largest];
            arr[largest] = swap;

            heapify(arr, n, largest);
        }
    }

    static void heapSort(int[] arr)
    {
        int n = arr.Length;

        for (int i = n / 2 - 1; i >= 0; i--)
            heapify(arr, n, i);

        for (int i = n - 1; i >= 0; i--)
        {
            int temp = arr[0];
            arr[0] = arr[i];
            arr[i] = temp;

            heapify(arr, i, 0);
        }
    }

    static void Main()
    {
        int[] arr = new int[15];
        int longitud = arr.Length;
        randomnumber(arr);
        Console.WriteLine("Array Original: ");
        printarray(arr);
        heapSort(arr);
        Console.WriteLine("Array Ordenado con Heap Sort: ");
        printarray(arr);
    }
}
